server/src/test/java/org/opensearch/node/NodeRoleSettingsTests.java

```diff
@@ -72,4 +72,10 @@ public void testUnknownNodeRoleOnly() {
         assertEquals(testRole, nodeRoles.get(0).roleName());
         assertEquals(testRole, nodeRoles.get(0).roleNameAbbreviation());
     }
+
+    public void testNodeRoleOfEmptyArrayNameShouldBeIgnored() {
+        String testRole = "[ ]";
+        Settings roleSettings = Settings.builder().put(NodeRoleSettings.NODE_ROLES_SETTING.getKey(), testRole).build();
```

Comment:
Since NODE_ROLES_SETTING is a list setting, `put` add value to the list and should not be treated as special string I believe